jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/java/SamInitProjectBuilderIntelliJ.kt

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.wizard.java
+
+import com.intellij.ide.projectWizard.ProjectTemplateList
+import com.intellij.ide.util.projectWizard.ModuleBuilder
+import com.intellij.ide.util.projectWizard.ModuleWizardStep
+import com.intellij.ide.util.projectWizard.WizardContext
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.module.JavaModuleType
+import com.intellij.openapi.module.ModuleType
+import com.intellij.openapi.projectRoots.JavaSdk
+import com.intellij.openapi.roots.ModifiableRootModel
+import com.intellij.openapi.roots.ui.configuration.ModulesProvider
+import com.jetbrains.python.module.PythonModuleType
+import com.jetbrains.python.sdk.PythonSdkType
+import icons.AwsIcons
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.ui.wizard.SAM_TEMPLATES
+import software.aws.toolkits.jetbrains.ui.wizard.SamModuleType
+import software.aws.toolkits.jetbrains.ui.wizard.SamProjectTemplateWrapper
+
+class SamInitModuleBuilder : ModuleBuilder() {
+    var runtime: Runtime = Runtime.UNKNOWN_TO_SDK_VERSION
+    lateinit var runtimeSelectionPanel: SamInitRuntimeSelectionPanel
+    lateinit var template: SamProjectTemplateWrapper
```

Comment:
so many `var`'s here! I get that it's a builder but do we have any control how the methods that mutate these properties get invoked?

Reply:
we do not :(
best i can do is change runtimeSelectionPanel to a val